pages/api/token/temporary.ts

```diff
@@ -18,37 +31,41 @@ import { MiddlewareProps } from "@lib/types";
  * @param email - the email for the user making the request, found in the body of the api request
  * @returns a promise
  */
-const handler = async (req: NextApiRequest, res: NextApiResponse, { formID }: MiddlewareProps) => {
-  try {
-    if (formID == undefined) {
-      return res.status(403).json({ error: "Invalid form request." });
-    }
-    if (!process.env.TOKEN_SECRET) {
-      return res.status(500).json({ error: "Server cannot respond at this time" });
-    }
-
-    const email = req.body["email"];
-    if (!email) {
-      res.status(400).json({ error: "Invalid payload" });
-    }
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  { formID, email }: MiddlewareProps
+): Promise<void> => {
+  assert(email); // Middleware function `requestContainsEmail` ensures that email is defined
+  assert(formID); // Middleware function `hasValidFormAccessToken` ensures that formID is defined
+  assert(process.env.TOKEN_SECRET); // Middleware function `hasValidFormAccessToken` ensures that TOKEN_SECRET is defined
 
+  try {
     const temporaryToken = createTemporaryToken(email, formID, process.env.TOKEN_SECRET);
     await updateTemporaryToken(temporaryToken, email, formID);
     await sendTemporaryTokenByEmail(email, temporaryToken);
 
+    await registerSuccessfulLoginAttempt(email);
+
     logMessage.info(`Temporary Token Requested: Form ID: ${formID} Email: ${email}`);
-    res.status(200).json({ message: "success" });
+    return res.status(200).json({ message: "success" });
   } catch (error) {
     logMessage.error("Failed to generate temporary token.");
     logMessage.error(error);
 
-    let errorMessage = "Malformed API Request";
-
-    if ((error as Error).message.includes("Could not send temporary token by email")) {
-      errorMessage = "GC Notify service failed to send temporary token";
+    if ((error as Error).message.includes("User is not found or not active")) {
```

Comment:
We may want to move this check up in the logic sequence so that if the error is only `User is not found or not active` it doesn't create an application error log.